torch/nn/modules/module.py

```diff
@@ -1144,10 +1144,24 @@ def to(self, *args, **kwargs):
                     "if a complex module does not work as expected.")
 
         def convert(t):
-            if convert_to_format is not None and t.dim() in (4, 5):
-                return t.to(device, dtype if t.is_floating_point() or t.is_complex() else None,
-                            non_blocking, memory_format=convert_to_format)
-            return t.to(device, dtype if t.is_floating_point() or t.is_complex() else None, non_blocking)
+            try:
+                if convert_to_format is not None and t.dim() in (4, 5):
+                    return t.to(
+                        device,
+                        dtype if t.is_floating_point() or t.is_complex() else None,
+                        non_blocking,
+                        memory_format=convert_to_format,
+                    )
+                return t.to(
+                    device,
+                    dtype if t.is_floating_point() or t.is_complex() else None,
+                    non_blocking,
+                )
+            except NotImplementedError as e:
+                if str(e) == "Cannot copy out of meta tensor; no data!":
+                    raise NotImplementedError(
+                        f"{e} Please use torch.nn.Module.to_empty() instead of torch.nn.Module.to()."
+                    ) from None
```

Comment:
Hm, why from None? I believe this should be 
```suggestion
                    ) from e
```

Reply:
```
if str(e) == "Cannot copy out of meta tensor; no data!":
raise NotImplementedError(
    "Please use torch.nn.Module.to_empty() instead of torch.nn.Module.to()."
) from None
```
gives

Traceback (most recent call last):
  File ".\torch\nn\modules\module.py", line 1159, in convert
    return t.to(device, dtype if t.is_floating_point() or t.is_complex() else None, non_blocking)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
NotImplementedError: Cannot copy out of meta tensor; no data!

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "test.py", line 19, in <module>
    m2.to("cpu")
  File ".\torch\nn\modules\module.py", line 1166, in to
    return self._apply(convert)
           ^^^^^^^^^^^^^^^^^^^^
  File ".\torch\nn\modules\module.py", line 833, in _apply
    param_applied = fn(param)
                    ^^^^^^^^^
  File ".\torch\nn\modules\module.py", line 1162, in convert
    raise NotImplementedError(
NotImplementedError: Please use torch.nn.Module.to_empty() instead of torch.nn.Module.to() when moving module from meta to a different device.
```

Reply:
```
if str(e) == "Cannot copy out of meta tensor; no data!":
raise NotImplementedError(
    f"{e} Please use torch.nn.Module.to_empty() instead of torch.nn.Module.to() when moving module from meta to a different device."
) from None
```

gives

```
Traceback (most recent call last):
  File "test.py", line 19, in <module>
    m2.to("cpu")
  File ".\torch\nn\modules\module.py", line 1166, in to
    return self._apply(convert)
           ^^^^^^^^^^^^^^^^^^^^
  File ".\torch\nn\modules\module.py", line 833, in _apply
    param_applied = fn(param)
                    ^^^^^^^^^
  File ".\torch\nn\modules\module.py", line 1162, in convert
    raise NotImplementedError(
NotImplementedError: Cannot copy out of meta tensor; no data! Please use torch.nn.Module.to_empty() instead of torch.nn.Module.to() when moving module from meta to a different device.
```

Reply:
The difference is the wether we show two message 

1. "Cannot copy out of meta tensor; no data!"
2. Please use torch.nn.Module.to_empty() instead of torch.nn.Module.to() when moving module from meta to a different device.

in one error or in two errors

Reply:
`from e` preserves the original exception context which is good but just makes the trace a bit messy. @mikaylagawarecki lmk WDYT. thanks

Reply:
Whoops, missed the fact that you kept the e at the start of the string.

Hm would prefer if we preserve the stack trace. Also, the current could would swallow the error here if str(e) was not " "Cannot copy out of meta tensor; no data!", please add the else: raise path as well!
```
if str(e) == "Cannot copy out of meta tensor; no data!":
    raise NotImplementedError(
        f"Please use torch.nn.Module.to_empty() instead of torch.nn.Module.to() when moving module from meta to a different device."
    ) from e
else:
     raise
```